File: D3/1206_View/1206.py
for test_case in range(1, 11):
    n = int(input())
    buildings = list(map(int, input().split()))
    ans = 0

    for i in range(2, n - 2):
        side_h = buildings[i-2:i+3]
        side_h.remove(buildings[i])
        diff_h = buildings[i] - max(side_h)
        ans += diff_h if diff_h > 0 else 0

    print(f"#{test_case} {ans}")

File: D3/1206_View/1206.cpp
#include<cstdio>
#include<iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(int argc, char** argv)
{
	// freopen("input.txt", "r", stdin);

	for(int test_case = 1; test_case <= 10; ++test_case)
	{
		int n;
		cin >> n;
		vector<int> buildings(n);

		for(int i = 0, h; i < n; i++) cin >> buildings[i];
		
		int ans = 0;
		for(int i = 2; i < n - 2; i++) {
			// Solution_1
			// vector<int> sub_buildings(buildings.begin() + (i - 2), buildings.begin() + (i + 3));
			// sub_buildings.erase(sub_buildings.begin() + 2);
			// int views = buildings[i] - *max_element(sub_buildings.begin(), sub_buildings.end());
			
			// Solution_2
			int views = max(max(buildings[i - 2], buildings[i - 1]), max(buildings[i + 1], buildings[i + 2]));
			views = buildings[i] - views;
			ans += (views > 0 ? views : 0);
		}

		cout << "#" << test_case << ' ' << ans << '\n';
	}
	return 0;
}